src/app/components/rating-picker/rating-picker.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, Input } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { BehaviorSubject, switchMap } from 'rxjs';
+import { RatingPostConfig } from 'src/app/types/rating-post-config';
+import { GuestSessionService } from 'src/app/services/guest-session.service';
+import { RatingService } from 'src/app/services/rating.service';
+import { NotificationsService } from 'src/app/services/notifications.service';
+import { CustomNotificationType } from 'src/app/types/notification';
+
+@Component({
+  selector: 'mdb-rating-picker',
+  templateUrl: './rating-picker.component.html',
+  styleUrls: ['./rating-picker.component.scss'],
+})
+export class RatingPickerComponent {
+  constructor(
+    private readonly ratingService: RatingService,
+    private readonly guestSessionService: GuestSessionService,
+    private readonly notificationsService: NotificationsService
+  ) {}
+
+  @Input() public initial = 5;
+  @Input() public ratingPostConfig?: RatingPostConfig;
+
+  public isSubmitting$ = new BehaviorSubject<boolean>(false);
+  public step = 0.5;
+  public min = this.step;
+  public max = 10;
+
+  public ratingValue = new FormControl(this.initial);
+
+  public submitRating() {
+    if (this.ratingPostConfig && this.ratingValue.value) {
+      const { id, type } = this.ratingPostConfig;
+      const value = this.ratingValue.value;
+
+      this.isSubmitting$.next(true);
+
+      this.guestSessionService
```

Comment:
It seems, this component shouldn't know anything about guest session and making request. Maybe we could provide this information from parent components?

Reply:
+ unsubscribe

Reply:
Updated: sessionId stored in localStorage only after successful rating POST + unsubscribed in onDestroy hook.